apps/web/src/hooks/useCreateWorkflowFromBlueprint.ts

```diff
@@ -0,0 +1,35 @@
+import slugify from 'slugify';
+import { useMutation, UseMutationOptions } from '@tanstack/react-query';
+import type { INotificationTemplate } from '@novu/shared';
+
+import { useCreateTemplateFromBlueprint } from '../api/hooks';
+import { getBlueprintTemplateById, getTemplateById } from '../api/notification-templates';
+import { TemplateCreationSourceEnum } from '../pages/templates/shared';
+import { getWorkflowBlueprintDetails } from '../utils';
+
+export const useCreateWorkflowFromBlueprint = (
+  options: UseMutationOptions<INotificationTemplate, any, { blueprintIdentifier: string }> = {}
+) => {
+  const { mutateAsync: createTemplateFromBlueprint } = useCreateTemplateFromBlueprint();
+
+  const { mutate: createWorkflowFromBlueprint, ...rest } = useMutation(async ({ blueprintIdentifier }) => {
+    const blueprintData = await getBlueprintTemplateById(blueprintIdentifier);
+    const { name: blueprintName } = getWorkflowBlueprintDetails(blueprintData.name);
+
+    try {
+      const workflowIdentifier = slugify(blueprintName, {
+        lower: true,
+        strict: true,
+      });
+
+      return await getTemplateById(workflowIdentifier);
+    } catch (error) {
+      return await createTemplateFromBlueprint({
+        blueprint: { ...blueprintData, name: blueprintName },
+        params: { __source: TemplateCreationSourceEnum.ONBOARDING_GET_STARTED },
+      });
```

Comment:
when the workflow doesn't exist, create a new one based on the blueprint

Reply:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** What should happen if this fails?‏

Reply:
if this fails, then the mutation will fail and in the result, we will show the error toast